### PR TITLE
dev-cmd/bump: Don't fall over when retrieving PRs 404s

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

- Instead, let the `pull_requests&.any?` check do its job and not show PRs that we couldn't find or fetch.
- In the `--debug` output, show the error message that we got from GitHub.
- Fixes https://github.com/Homebrew/brew/issues/16504.

### CODE REVIEW

**Comment by issyl0** [comment]
Wait, it's not an auth error in the original issue, I'm being an idiot. That's the 404 message instead.

**Comment by Bo98** [comment]
Bug report looks to be a ValidationFailedError rather than a PageNotFoundError

**Comment by issyl0** [comment]
And this is why we have code review. 😅 Thanks, Bo, you were correct!

File: Library/Homebrew/dev-cmd/bump.rb [hunk 1]
```diff
@@ -288,7 +288,12 @@ def livecheck_result(formula_or_cask)
   }
   def retrieve_pull_requests(formula_or_cask, name, state:, version: nil)
     tap_remote_repo = formula_or_cask.tap&.remote_repo || formula_or_cask.tap&.full_name
-    pull_requests = GitHub.fetch_pull_requests(name, tap_remote_repo, state: state, version: version)
+    begin
+      pull_requests = GitHub.fetch_pull_requests(name, tap_remote_repo, state: state, version: version)
+    rescue GitHub::API::ValidationFailedError => e
+      odebug "Error fetching pull requests for #{formula_or_cask} #{name}: #{e}"
+    end
+
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump.rb:296]:
```suggestion
    pull_requests = begin
      GitHub.fetch_pull_requests(name, tap_remote_repo, state: state, version: version)
    rescue GitHub::API::ValidationFailedError => e
      odebug "Error fetching pull requests for #{formula_or_cask} #{name}: #{e}"
      nil
    end

```

feels a little more explicit

**Review by MikeMcQuaid — APPROVED** [review]
Looks good! One optional suggestion and then can self-merge.

**Pull request merged**